README.md

```diff
@@ -86,38 +156,159 @@ bool_value = client.fetch_boolean_value(
 
-The `NoOpProvider` is an example of a minimalist provider. The `InMemoryProvider` is a provider that can be initialized with flags and used to store flags in process. For complete documentation on the Provider interface, visit: https://openfeature.dev/specification/sections/providers.
+### Domains
 
-In addition to the `fetch_*` methods, providers can optionally implement lifecycle methods that are invoked when the underlying provider is switched out. For example:
+Clients can be assigned to a domain. A domain is a logical identifier which can be used to associate clients with a particular provider.
+If a domain has no associated provider, the default provider is used.
+
+```ruby
+OpenFeature::SDK.configure do |config|
+  config.set_provider(OpenFeature::SDK::Provider::NoOpProvider.new, domain: "legacy_flags")
+end
+
+# Create a client for a different domain, this will use the provider assigned to that domain
+legacy_flag_client = OpenFeature::SDK.build_client(domain: "legacy_flags")
+```
+
+### Eventing
+
+Coming Soon!
+
+<!-- Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
+Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
+Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
+
+Please refer to the documentation of the provider you're using to see what events are supported. -->
+
+<!-- TODO: code example of a PROVIDER_CONFIGURATION_CHANGED event for the client and a PROVIDER_STALE event for the API -->
+
+### Shutdown
+
+Coming Soon!
+
+<!-- TODO The OpenFeature API provides a close function to perform a cleanup of all registered providers.
+This should only be called when your application is in the process of shutting down.
+
+```ruby
+class MyProvider
+  def shutdown
+    # Perform any shutdown/reclamation steps with flag management system here
+    # Return value is ignored
+  end
+end
+``` -->
+
+### Transaction Context Propagation
+
+Coming Soon!
+
+<!-- Transaction context is a container for transaction-specific evaluation context (e.g. user id, user agent, IP).
+Transaction context can be set where specific data is available (e.g. an auth service or request handler) and by using the transaction context propagator it will automatically be applied to all flag evaluations within a transaction (e.g. a request or thread). -->
+
+<!-- TODO: code example for global shutdown -->
+
+## Extending
+
+### Develop a provider
+
+To develop a provider, you need to create a new project and include the OpenFeature SDK as a dependency.
+This can be a new repository or included in [the existing contrib repository](https://github.com/open-feature/ruby-sdk-contrib) available under the OpenFeature organization.
+You’ll then need to write the provider by implementing the `Provider` duck.
```

Comment:
Is it really called a duck? 😄

Reply:
Oh yea, [duck-typing](https://en.wikipedia.org/wiki/Duck_typing) is all the rage in Ruby

Reply:
Ah, understand. I'm familiar with duck typing but I hadn't seen it referred to as duck. Thanks for the clarification 😀